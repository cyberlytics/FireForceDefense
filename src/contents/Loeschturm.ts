import Content from '../model/Content';
import ContentCategory from '../model/ContentCategory';
import type Cell from '../model/Cell';

export default class Loeschturm extends Content {
    readonly description = 'Loeschturm: Description';
    readonly name = 'Loeschturm';
    readonly id = 'Loeschturm';
    readonly buildCosts: number = 15;
    readonly damageMax = Content.DAMAGE_MAX_LOW;
    readonly fireGrowChance = 0;
    readonly fireGrowAmount = 0;
    readonly removeCosts: number = -10;
    readonly extinguishRange = 1;
    readonly extinguishRate = 2;
    readonly contentCategory = ContentCategory.CONTENT_EXTINGUISHER;

    isPlaceableOn(cell: Cell): boolean {
        return cell.groups.includes('land');
    }
}
