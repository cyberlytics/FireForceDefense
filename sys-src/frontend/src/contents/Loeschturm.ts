import Content from '@model/Content';
import ContentCategory from '@model/ContentCategory';
import type Cell from '@model/Cell';

export default class Loeschturm extends Content {
    readonly description =
        'Fire-fighting tower: A big tower with a built-in water pump and a raised fire extinguishing range.';
    readonly name = 'Loeschturm';
    readonly id = 'Loeschturm';
    readonly buildCosts: number = 25;
    readonly damageMax = Content.DAMAGE_MAX_LOW;
    readonly fireGrowChance = 0;
    readonly fireGrowAmount = 0;
    readonly removeCosts: number = -15;
    readonly extinguishRange = 2;
    readonly extinguishRate = 1;
    readonly extinguishChance = 0.5;
    readonly contentCategory = ContentCategory.CONTENT_EXTINGUISHER;

    isPlaceableOn(cell: Cell): boolean {
        return cell.groups.includes('land');
    }
}
