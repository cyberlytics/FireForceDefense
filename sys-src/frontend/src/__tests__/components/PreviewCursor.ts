import { mount } from '@vue/test-utils';
import PreviewCursor from '@components/previewCursor.vue';
import Loeschturm from '@contents/Loeschturm';

describe('PreviewCursor.vue', () => {
    test('empty preview cursor matches snapshot', () => {
        const wrapper = mount(PreviewCursor, {
            propsData: {
                contentToBuild: null,
                mouseX: 13,
                mouseY: 37,
                removeMode: false,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    test('preview cursor with content to build matches snapshot', () => {
        const wrapper = mount(PreviewCursor, {
            propsData: {
                contentToBuild: Loeschturm,
                mouseX: 13,
                mouseY: 37,
                removeMode: false,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    test('preview cursor in remove mode matches snapshot', () => {
        const wrapper = mount(PreviewCursor, {
            propsData: {
                contentToBuild: null,
                mouseX: 13,
                mouseY: 37,
                removeMode: true,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    test('contentToBuild takes precedence over removeMode', () => {
        const wrapper = mount(PreviewCursor, {
            propsData: {
                contentToBuild: Loeschturm,
                mouseX: 13,
                mouseY: 37,
                removeMode: true,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
