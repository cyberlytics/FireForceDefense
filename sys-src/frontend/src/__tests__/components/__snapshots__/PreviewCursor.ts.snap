// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewCursor.vue contentToBuild takes precedence over removeMode 1`] = `
<svg
  class="position-absolute"
  height="100px"
  preserveAspectRatio="xMidYMid meet"
  style="opacity: 0.7; pointer-events: none; left: -37px; top: -13px;"
  viewBox="0 0 100 100"
  width="100px"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <rect
    fill="url(#content-Loeschturm)"
    height="100"
    width="100"
    x="0"
    y="0"
  />
</svg>
`;

exports[`PreviewCursor.vue empty preview cursor matches snapshot 1`] = `<!---->`;

exports[`PreviewCursor.vue preview cursor in remove mode matches snapshot 1`] = `
<svg
  class="position-absolute"
  height="100px"
  preserveAspectRatio="xMidYMid meet"
  style="opacity: 0.7; pointer-events: none; left: -37px; top: -13px;"
  viewBox="0 0 100 100"
  width="100px"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <rect
    fill="url(#utility-Abbauen)"
    height="100"
    width="100"
    x="0"
    y="0"
  />
</svg>
`;

exports[`PreviewCursor.vue preview cursor with content to build matches snapshot 1`] = `
<svg
  class="position-absolute"
  height="100px"
  preserveAspectRatio="xMidYMid meet"
  style="opacity: 0.7; pointer-events: none; left: -37px; top: -13px;"
  viewBox="0 0 100 100"
  width="100px"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <rect
    fill="url(#content-Loeschturm)"
    height="100"
    width="100"
    x="0"
    y="0"
  />
</svg>
`;
